run.py

```diff
@@ -172,13 +172,13 @@ def run(args):
     sys.path.append(os.path.abspath('tests'))
     tests = suite.get('tests')
     tcs = []
-    jenkins_rc = 0
+    jenkins_rc = TestStatus.FAILED
     test_data = dict()
     if cluster_name:
         test_data['cluster-name'] = cluster_name
 
     for test in tests:
-        config = test.get('config', {})
+        config = test.get('test').get('config', {})
```

Comment:
if the test won't be defined here test.get('test') and it will return None as default value for get you will fail with next one .get as None type has no method called get.

Reply:
Its not possible to reach test which is None, since tests is a list of test from yaml.

Reply:
@petr-balogh is there a better way for me to access that sub-key without hitting an exception if the initial one doesn't exist? It's really not a big deal here because like vasu said the only way for us to hit that is for you to be using a malformed suite file. However, I would like to know if there's a better approach to this general issue.

Reply:
I can only think of nested if checks, but really dont see much value trying to check for malformed item which can be a dict or something else too. The review or test runs should throw them out.

Reply:
I see, thanks for explenation.  @clacroix12 not sure about elegant way.. I usually used .get('key', {}).get('key2', default) .